File: jet-lag/jetlag_test.go
package jetlag_test

import (
	"testing"

	jetlag "github.com/hsmtkk/fuzzy-giggle/jet-lag"
	"github.com/stretchr/testify/assert"
)

func Test1(t *testing.T) {
	records := []jetlag.Record{
		jetlag.NewRecord(jetlag.NewHour(7), jetlag.NewHour(5), jetlag.NewHour(12)),
		jetlag.NewRecord(jetlag.NewHour(10), jetlag.NewHour(6), jetlag.NewHour(20)),
		jetlag.NewRecord(jetlag.NewHour(12), jetlag.NewHour(3), jetlag.NewHour(8)),
	}
	p := jetlag.NewProblem(records)
	want := jetlag.NewAnswer(jetlag.NewHour(20), jetlag.NewHour(31))
	got := jetlag.Solve(p)
	assert.Equal(t, want, got)
}

File: jet-lag/jet_lag_test.py
import unittest

import jet_lag


class TestJetLag(unittest.TestCase):
    def test1(self) -> None:
        records: list[jet_lag.Record] = [
            jet_lag.Record(7, 5, 12),
            jet_lag.Record(10, 6, 20),
            jet_lag.Record(12, 3, 8),
        ]
        want = jet_lag.Answer(20, 31)
        got = jet_lag.solve(records)
        self.assertEqual(want, got)

    def test2(self) -> None:
        records: list[jet_lag.Record] = [
            jet_lag.Record(1, 1, 23),
            jet_lag.Record(23, 1, 1),
        ]
        want = jet_lag.Answer(3, 47)
        got = jet_lag.solve(records)
        self.assertEqual(want, got)

    def test3(self) -> None:
        records: list[jet_lag.Record] = [
            jet_lag.Record(5, 4, 20),
        ]
        want = jet_lag.Answer(13, 13)
        got = jet_lag.solve(records)
        self.assertEqual(want, got)


if __name__ == "__main__":
    unittest.main()
